src/api/forms/service.js

```diff
@@ -189,6 +197,55 @@ export async function updateDraftFormDefinition(formId, definition, author) {
   }
 }
 
+/**
+ * @param {string} formId - ID of the form
+ * @param {Partial<FormMetadataInput>} formUpdate - full JSON form definition
+ * @returns {Promise<string>}
+ */
+export async function updateFormMetadata(formId, formUpdate) {
+  logger.info(`Updating form metadata for form ID ${formId}`)
+
+  try {
+    // Get the form metadata from the db
+    const form = await getForm(formId)
+
+    if (form.live && 'title' in formUpdate) {
+      throw Boom.badRequest(
+        `Form with ID '${formId}' is live so 'title' cannot be updated`
+      )
+    }
+
+    let updatedSlug
+
+    const updatedForm = {
+      slug: form.slug,
+      ...formUpdate
+    }
+
+    if (formUpdate.title) {
+      updatedSlug = slugify(formUpdate.title)
+    } else {
+      updatedSlug = form.slug
+    }
```

Comment:
This has changed quite a bit since we last spoke, what made you change it? We're now returning an updated slug but not setting this in the database.

Your previous approach was almost there.

Taking Dave's comment as inspiration:

```javascript 

const updatedForm = { ...formUpdate }

if (formUpdate.title) {
  updatedForm.slug = slugify(formUpdate.title)
}
```

with something like this, the return would be something like `return updatedForm.slug ?? form.slug`

Reply:
If I remember correct, when I tried above I get 

```
Property 'slug' does not exist on type '{ title?: string | undefined; organisation?: string | undefined; teamName?: string | undefined; teamEmail?: string | undefined; }'
```

after which I should have done this - 

```
 const updatedForm = {
      slug: form.slug,
      ...formUpdate
    }

    if (formUpdate.title) {
      updatedForm.slug = slugify(formUpdate.title)
    }
```

and return updatedForm.slug

this should work I think

Reply:
if I spy'd on formMetadata.update and test on args, I would have caught this issue in tests

Reply:
Ah that's because no type was set for `{ ...formUpdate }`

The compiler needs to know in advance what the future (optional) keys might be

```patch
+   /** @satisfies {Partial<FormMetadataDocument>} */
    const updatedForm = {
```